src/freenet/support/Fields.java

```diff
@@ -721,10 +721,15 @@ public static int parseInt(String s, Dimension dimension) throws NumberFormatExc
 			case SIZE:
 				return parseInt(s);
 			case DURATION:
-				long durationInMillis = TimeUtil.toMillis(s);
-				if ((int) durationInMillis == durationInMillis)
-					return (int) durationInMillis;
-				throw new ArithmeticException("integer overflow");
+				try {
+					long durationInMillis = TimeUtil.toMillis(s);
+					if ((int) durationInMillis == durationInMillis) {
+						return (int) durationInMillis;
+					}
+					throw new ArithmeticException("integer overflow");
+				} catch (NumberFormatException e) {
+					return parseInt(s);
```

Comment:
shouldn’t the NumberFormatException be caught where the method is used — and for example lead to ignoring he new value?

Reply:
yes, that’s much better. the Fields class is not relevant to this issue. and the fix code is next to the cause.